### ISSUE TITLE
Warehouse - Amazon Shipment 2D Barcode Generation Page

### ISSUE THREAD

**Issue by snaiperskaya96** [snaiperskaya96/test-import-repo#183]
https://trello.com/c/y6JwT2OA/593-warehouse-amazon-shipment-2d-barcode-generation-page

Must be merged first:

https://trello.com/c/um0EJBnY/616-warehouse-add-intermediate-barcode-to-goods-in-page

https://trello.com/c/scEry3wn/596-amazon-shipment-creator-part-1

---

A new warehouse page is to be created called "Shipments".

This pages top level aim is to generate the required 2D box barcode for Amazon which details out the ASIN, QTY, EXP DATE and Amazon PO number.

The 2D Barcode is created from 2 sources of Data
1. Amazon Shipment ID (AMZ PO Number)
2. Box contents details

example:

```
AMZN,PO:FBA23JFC2G,ASIN:B00H1BPQ1Y,QTY:12,EXP:140301
```

The problem is that the box contents information is recorded at point of goods in / barcoding and we don't yet know the AMZN Shipment ID at this point in time.

**The solution is as follows:**

For the moment Ryan will still create shipments manually in the background. These shipments will then exist on Amazon.

We will use the ListInboundShipments and ListInboundShipmentItems to maintain a database table of open (working) shipments SKU's ASINS expected QTY's and Amazon PO numbers. A separate task should be created for this.

http://docs.developer.amazonservices.com/en_UK/fba_inbound/FBAInbound_ListInboundShipmentItems.html

The UI will have a single entry area for the scanner. When the above INTERMEDIATE barcode is scanned the ASIN/SKU is cross referenced to an open shipment (working) the QTY/SKU's cross checked. Then the 2D barcode is generated and printed.

The UI will display all the expected SKU's and Quantities for a shipment and background will go green when QTY/Labels have been printed

See example graphic for UI
